contracts/strategies/euler/IdleEulerStakingStrategy.sol

```diff
@@ -0,0 +1,227 @@
+    // Public methods
+    // ###################
+
+    /// @dev msg.sender should approve this contract first to spend `_amount` of `token`
+    /// @param _amount amount of `token` to deposit
+    /// @return shares strategyTokens minted
+    function deposit(uint256 _amount) external override onlyIdleCDO returns (uint256 shares) {
+        if (_amount != 0) {
+            uint256 eTokenBalanceBefore = eToken.balanceOf(address(this));
+            // Send tokens to the strategy
+            underlyingToken.safeTransferFrom(msg.sender, address(this), _amount);
+
+            eToken.deposit(SUB_ACCOUNT_ID, _amount);
+
+            // Mint shares 1:1 ratio
+            shares = eToken.balanceOf(address(this)) - eTokenBalanceBefore;
+            if (address(stakingRewards) != address(0)) {
+                stakingRewards.stake(shares);
+            }
+
+            _mint(msg.sender, shares);
+        }
+    }
+
+    function redeemRewards(bytes calldata data)
+        public
+        override
+        onlyIdleCDO
+        nonReentrant
+        returns (uint256[] memory rewards)
+    {
+        rewards = _redeemRewards(data);
+    }
+
+    /// @dev msg.sender should approve this contract first to spend `_amount` of `strategyToken`
+    /// @param _shares amount of strategyTokens to redeem
+    /// @return amountRedeemed  amount of underlyings redeemed
+    function redeem(uint256 _shares) external override onlyIdleCDO returns (uint256 amountRedeemed) {
+        if (_shares != 0) {
+            _burn(msg.sender, _shares);
+            // Withdraw amount needed
+            amountRedeemed = _withdraw((_shares * price()) / EXP_SCALE, msg.sender);
+        }
+    }
+
+    /// @notice Redeem Tokens
+    /// @param _amount amount of underlying tokens to redeem
+    /// @return amountRedeemed Amount of underlying tokens received
+    function redeemUnderlying(uint256 _amount) external override onlyIdleCDO returns (uint256 amountRedeemed) {
+        uint256 _shares = (_amount * EXP_SCALE) / price();
+        if (_shares != 0) {
+            _burn(msg.sender, _shares);
+            // Withdraw amount needed
+            amountRedeemed = _withdraw(_amount, msg.sender);
+        }
+    }
+
+    // ###################
+    // Internal
+    // ###################
+
+    /// @dev Unused but needed for BaseStrategy
+    function _deposit(uint256 _amount) internal override returns (uint256 amountUsed) {}
+
+    /// @param _amountToWithdraw in underlyings
+    /// @param _destination address where to send underlyings
+    /// @return amountWithdrawn returns the amount withdrawn
+    function _withdraw(uint256 _amountToWithdraw, address _destination)
+        internal
+        override
+        returns (uint256 amountWithdrawn)
+    {
+        IEToken _eToken = eToken;
+        IERC20Detailed _underlyingToken = underlyingToken;
+
+        // Unstake from StakingRewards
+        stakingRewards.withdraw(_eToken.convertUnderlyingToBalance(_amountToWithdraw));
```

Comment:
we should add `if else` statement for checking `stakingRewards` is non-zero address.

Reply:
ok I've added the guard here and in `_redeemRewards` and removed the `require` in `setStakingRewards`. I initially thought we should never allow `stakingRewards` to be address(0) but actually it could be handy to skip all rewards logic if eg staking is disabled